core/controllers/profile.py

```diff
@@ -89,8 +89,15 @@ def get(self, username):
             edited_exp_summary_dicts = (
                 summary_services.get_displayable_exp_summary_dicts_matching_ids(
                     user_contributions.edited_exploration_ids))
+            ### function to determine whether current user is same as user on page ###
+            if (self.username == username):
+                profile_is_of_current_user = True
+            if (self.username != username):
+                profile_is_of_current_user = False
+
```

Comment:
thanks Sean! 

Python tends to be quite lenient in type (especially true with booleans), but it is a double edged sword.